torchgeo/datasets/geonrw.py

```diff
@@ -0,0 +1,334 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""GeoNRW dataset."""
+
+import os
+from collections.abc import Callable
+from glob import glob
+from typing import ClassVar
+
+import matplotlib
+import matplotlib.cm
+import matplotlib.colors as mcolors
+import matplotlib.pyplot as plt
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+from torchvision import transforms
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import Path, download_and_extract_archive, extract_archive
+
+
+class GeoNRW(NonGeoDataset):
+    """GeoNRW dataset.
+
+    This datasets contains RGB, DEM and segmentation label data from North Rhine-Westphalia, Germany.
+
+    Dataset features:
+
+    * 7298 training and 485 test samples
+    * RGB images, 1000x1000px normalized to [0, 1]
+    * DEM images, unnormalized
+    * segmentation labels
+
+    Dataset format:
+
+    * RGB images are three-channel jp2
+    * DEM images are single-channel tif
+    * segmentation labels are single-channel tif
+
+    Dataset classes:
+
+    0. background
+    1. forest
+    2. water
+    3. agricultural
+    4. residential,commercial,industrial
+    5. grassland,swamp,shrubbery
+    6. railway,trainstation
+    7. highway,squares
+    8. airport,shipyard
+    9. roads
+    10. buildings
+
+    Additional information about the dataset can be found `on this site <https://ieee-dataport.org/open-access/geonrw>`__.
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://ieeexplore.ieee.org/document/9406194
+
+
+    .. versionadded:: 0.6
+    """
+
+    # Splits taken from https://github.com/gbaier/geonrw/blob/ecfcdbca8cfaaeb490a9c6916980f385b9f3941a/pytorch/nrw.py#L48
+
+    splits = ('train', 'test')
+
+    train_list: ClassVar[list[str]] = [
```

Comment:
Use a tuple here and elsewhere instead of a list so you can avoid ClassVar

Reply:
But then I get mypy complaining about assignin tuples of different length `torchgeo/datasets/geonrw.py:198: error: Incompatible types in assignment (expression has type "tuple[str, ...]", variable has type "tuple[str, str, str]") [assignment]`

Reply:
Bleh. I encountered this a few times in #2218. You can solve this using:
```suggestion
    train_list: tuple[str, ...] = (
```
to make it clear that the length of the tuple is meaningless/variable.